### PR TITLE
ci: 🎡 run integ tests in headless on windows as well

### PR DESCRIPTION
**Pull request by maniator**
---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by qhanam** [comment]
Hi @maniator what is the motivation for this change?

Running the windows integ test headless might speed up the workflow (and maybe reduce the windows test flakiness), but is there a benefit to running both the headless and non-headless windows integ tests?

**Comment by maniator** [comment]
> Hi @maniator what is the motivation for this change?
> 
> Running the windows integ test headless might speed up the workflow (and maybe reduce the windows test flakiness), but is there a benefit to running both the headless and non-headless windows integ tests?

Really prob better to run both in headless

But the headless config does not include Microsoft edge (which we can probably just update and create a Windows edge config and a unix edge config

**Comment by qhanam** [comment]
So with this change, the windows CI workflow will be:
1. Run integ tests with UI
   1. Chrome
   1. Firefox
   1. Edge
1. Run integ tests in headless
   1. Chrome
   1. Firefox

> But the headless config does not include Microsoft edge (which we can probably just update and create a Windows edge config and a unix edge config

Should we do this before merging so we don't re-run the tests? Or would you rather merge this and make the change later?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -73,13 +73,41 @@ jobs:
...
+        env:
+            NODE_OPTIONS: --max-http-header-size=80000
+        strategy:
+            fail-fast: false
+            matrix:
+                os: [ubuntu-latest, windows-latest]
+
+        steps:
+            - name: Checkout AWS RUM Web Client Repository
+              uses: actions/checkout@v2
```
Comment (piwysocki) [inline .github/workflows/ci.yaml:94]:
```suggestion
              uses: actions/checkout@v3
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -73,13 +73,41 @@ jobs:
...
+
+            - name: Setup Node
+              uses: actions/setup-node@v2
```
Comment (piwysocki) [inline .github/workflows/ci.yaml:97]:
```suggestion
              uses: actions/setup-node@v3
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -73,13 +73,41 @@ jobs:
...
+              with:
+                  node-version: '16.x'
+
+            - name: Cache NPM modules
+              uses: actions/cache@v2
```
Comment (piwysocki) [inline .github/workflows/ci.yaml:102]:
```suggestion
              uses: actions/cache@v3
```

**Comment by qhanam** [comment]
Closing, as Windows browser integ tests are now running headless.

**Pull request closed without merging**